data/moves.ts

```diff
@@ -17445,6 +17964,25 @@ export const BattleMovedex: {[moveid: string]: MoveData} = {
 		target: "normal",
 		type: "Steel",
 	},
+	steelroller: {
+		num: 798,
+		accuracy: 100,
+		basePower: 130,
+		category: "Physical",
+		desc: "Ends the effects of Electric Terrain, Grassy Terrain, Misty Terrain, and Psychic Terrain. This move fails if there is not a terrain active.",
+		shortDesc: "Fails if no terrain. Ends terrain.",
+		name: "Steel Roller",
+		pp: 5,
+		priority: 0,
+		flags: {contact: 1, protect: 1, mirror: 1},
+		onPrepareHit() {
+			if (!this.field.getTerrain()) return false;
+			this.field.clearTerrain();
```

Comment:
Should this still clear the terrain if it subsequently fails to execute for some reason?

Reply:
I'm not sure I understand what you mean

Reply:
`onPrepareHit` runs very early in the lifetime of a move, before such things as checking for Protect. It's possible that the terrain doesn't get cleared if the move doesn't hit its target.

Reply:
oh true